flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/common/RMQConnectionConfig.java

```diff
@@ -29,21 +29,24 @@
 import java.security.KeyManagementException;
 import java.security.NoSuchAlgorithmException;
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 
 /**
  * Connection Configuration for RMQ. If {@link Builder#setUri(String)} has been set then {@link
  * RMQConnectionConfig#RMQConnectionConfig(String, Integer, Boolean, Boolean, Integer, Integer,
- * Integer, Integer, Integer)} will be used for initialize the RMQ connection or {@link
+ * Integer, Integer, Integer, Integer)} will be used for initialize the RMQ connection or {@link
  * RMQConnectionConfig#RMQConnectionConfig(String, Integer, String, String, String, Integer,
- * Boolean, Boolean, Integer, Integer, Integer, Integer, Integer)} will be used for initialize the
- * RMQ connection
+ * Boolean, Boolean, Integer, Integer, Integer, Integer, Integer, Integer)} will be used for
+ * initialize the RMQ connection
  */
 public class RMQConnectionConfig implements Serializable {
 
     private static final long serialVersionUID = 1L;
 
     private static final Logger LOG = LoggerFactory.getLogger(RMQConnectionConfig.class);
 
+    private static final int DEFAULT_DELIVERY_TIMEOUT = 30000;
```

Comment:
Why is the default value of delivery timeout is 30 seconds?

Reply:
We decided in the discussion under the related ticket (https://issues.apache.org/jira/browse/FLINK-22698) to set a sensible default that will resolve the potential problem that a user may encounter while stopping the job (the other option is to just leave the current problematic behavior by setting in to 0).
There's nothing special about 30s, but I would say it should be lower than 60s as this is the default timeout for the CLI client stop command.

Reply:
My point of concern is why the default value is 30 seconds, and what is the basis for the default value of 30 seconds.

Reply:
Yes, I understand. The issue described FLINK-22698 causes the job to never stop (gracefully via stop command) if auto-ack is disabled and there is no new data. So, the current behavior (no timeout) is causing trouble (that's why its better to set some default here).
IMO the upper bound is dictated by the default Flink _client.timeout_ (60s) - RMQ connector should not be the cause of CLI timeouts. The lower bound is related to CPU utilization - if set to some really low values (such as 1ms), it could start to have a noticeable effect (redundant queue check every 1ms).
Again, there is nothing special with the 30s (e.g. we can set it to 15s as well), but it **fixes the issue** and meets the above criteria.

Reply:
Thanks for your detailed explanation. I got your point for the delivery timeout and makes sense to me.

Reply:
Great, thanks. I updated the PR description with the details above if anyone else was looking for a rationale